### PR TITLE
[VL] Add Scala 2.13 support

### PR DESCRIPTION
**Pull request by Preetesh2110**
## What changes were proposed in this pull request?

 Add Scala 2.13 support

(Fixes: \#6316)

## How was this patch tested?

Build locally

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by surnaik** [comment]
@zhztheplayer Could you please review this PR? Thanks!

**Comment by LuciferYang** [comment]
Prior to Spark 4.0, Scala 2.13 was not the default Scala version, and I personally do not recommend supporting it for Spark 3.x.

**Comment by surnaik** [comment]
Thanks @LuciferYang, I think Scala 2.13 support is already added for CH backend, plan was to extend to VL backend as well, since the build fails when you enable `-Pbackends-velox -Pscala-2.13`, that was the plan, does that make sense.

FYI, the support for Scala 2.13 was added here already - #6311 

cc: @zhztheplayer

**Comment by surnaik** [comment]
@Preetesh2110 please take a look at failing CI and maybe also add a CI option to build with Scala 2.13

**Comment by LuciferYang** [comment]
> Thanks @LuciferYang, I think Scala 2.13 support is already added for CH backend, plan was to extend to VL backend as well, since the build fails when you enable `-Pbackends-velox -Pscala-2.13`, that was the plan, does that make sense.
> 
> FYI, the support for Scala 2.13 was added here already - #6311
> 
> cc: @zhztheplayer

Got

**Comment by zhztheplayer** [comment]
Do we need to have a test for Scala 2.13 (with Spark 3.5 probably) ? https://github.com/apache/incubator-gluten/blob/main/.github/workflows/velox_docker.yml

**Comment by Preetesh2110** [comment]
Thanks for the feedback. I have added a workflow @zhztheplayer.

**Comment by zhztheplayer** [comment]
Run Gluten Clickhouse CI

File: .github/workflows/velox_docker.yml
Comment (zhztheplayer) [inline .github/workflows/velox_docker.yml:1097]: This could make job `run-spark-test-spark35` much slower.

If so, probably we should add a job `run-spark-test-spark35-scala213`. cc @zhouyuan @PHILO-HE

Comment (Preetesh2110) [reply .github/workflows/velox_docker.yml]: Done. Thanks

**Comment by surnaik** [comment]
@Preetesh2110 Please fix the CI failures

**Comment by zhztheplayer** [comment]
@Preetesh2110 

Thank you for keeping working on this!

BTW, in case you may wait for CI to be triggered, you can open a PR in your own repo to run CI if you want.

**Comment by zhztheplayer** [comment]
> > @Preetesh2110
> > Thank you for keeping working on this!
> > BTW, in case you may wait for CI to be triggered, you can open a PR in your own repo to run CI if you want.
> 
> Hey @zhztheplayer thanks for your message, how can I trigger the CIs, I have created a PR in my own repo [Preetesh2110#1](https://github.com/Preetesh2110/incubator-gluten/pull/1).

You can sync gluten's main branch to your own main branch, then open a PR targeting to your own main branch. The CI should be directly runnable.

Anyway you can still update here if it's OK for you to wait for CI to be triggered.

Once you have one PR merged you will not get blocked in your subsequent PRs.

**Comment by zhouyuan** [comment]
@Preetesh2110 FYI, below mvn command can fix the style issues so this will not stop CI
https://github.com/apache/incubator-gluten/blob/main/docs/developers/NewToGluten.md?plain=1#L111

thanks, -yuan

**Review by surnaik — APPROVED** [review]
LGTM - Non binding

**Comment by surnaik** [comment]
@zhztheplayer please take a look, thanks! 
cc: @weiting-chen

File: .scalafmt.conf
Comment (zhztheplayer) [inline .scalafmt.conf:4]: Any reason updating the version?

**Review by zhztheplayer — APPROVED** [review]
+1 overall. I didn't try with the feature yet but assume CI should cover the changes.

Thank your for the effort @Preetesh2110

File: .scalafmt.conf
Comment (surnaik) [reply .scalafmt.conf:4]: Thanks @zhztheplayer for reviewing. When using the older version, the CI ran into exception
```
2024-08-08T00:57:49.5831318Z [INFO] --- spotless-maven-plugin:2.27.2:check (spotless-check) @ gluten-ut-spark35 ---
2024-08-08T00:57:50.3039149Z [ERROR] Step 'scalafmt' found problem in 'src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala':
2024-08-08T00:57:50.3040518Z null
2024-08-08T00:57:50.3040881Z java.lang.StackOverflowError
2024-08-08T00:57:50.3101797Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3102978Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3104855Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3108247Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3110226Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3113485Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3116925Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3125982Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3132026Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3133287Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3134730Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3136020Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3137364Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3138459Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3139722Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3141082Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3142179Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3143475Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
...
...
2024-08-08T00:57:50.3676975Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3679088Z Exception in thread "main" java.lang.StackOverflowError
2024-08-08T00:57:50.3680068Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
2024-08-08T00:57:50.3681434Z 	at scala.meta.transversers.Traverser.apply(Traverser.scala:4)
2024-08-08T00:57:50.3682658Z 	at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply(FormatWriter.scala:1568)
2024-08-08T00:57:50.3683915Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
2024-08-08T00:57:50.3684924Z 	at scala.meta.transversers.Traverser.apply(Traverser.scala:4)
2024-08-08T00:57:50.3686232Z 	at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply(FormatWriter.scala:1568)
2024-08-08T00:57:50.3687447Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
2024-08-08T00:57:50.3688408Z 	at scala.meta.transversers.Traverser.apply(Traverser.scala:4)
2024-08-08T00:57:50.3689522Z 	at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply(FormatWriter.scala:1568)
2024-08-08T00:57:50.3690706Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
```

This happened consistently and multiple tries of
`mvn spotless:apply -P....` didn't help, we tried both Velox and Clickhouse profiles with UTs.

Upgrading the version helped.

Comment (surnaik) [reply .scalafmt.conf:4]: Here's the failure CI link - https://github.com/apache/incubator-gluten/actions/runs/10280644913/job/28490950203?pr=6326

**Pull request merged**
